src/Raven.Server/Smuggler/Documents/StreamSource.cs

```diff
@@ -1125,20 +1134,58 @@ private IEnumerable<DocumentItem> ReadDocuments(INewDocumentActions actions = nu
+                        storeDocsForLaterMatchWithAttachments.Add(modifier.Id.ToLowerInvariant(), new DocumentItem
                         {
-                            Data = data,
-                            Id = context.GetLazyString(modifier.Id),
-                            ChangeVector = modifier.ChangeVector,
-                            Flags = modifier.Flags,
-                            NonPersistentFlags = modifier.NonPersistentFlags,
-                            LastModified = modifier.LastModified ?? _database.Time.GetUtcNow(),
-                        },
-                        Attachments = attachments
-                    };
-                    attachments = null;
+                            Document = new Document
+                            {
+                                Data = data,
+                                Id = context.GetLazyString(modifier.Id),
+                                ChangeVector = modifier.ChangeVector,
+                                Flags = modifier.Flags,
+                                NonPersistentFlags = modifier.NonPersistentFlags,
+                                LastModified = modifier.LastModified ?? _database.Time.GetUtcNow(),
+                            },
+                            Attachments = null
+                        });
+                    }
+                    else
+                    {
+                        yield return new DocumentItem
+                        {
+                            Document = new Document
+                            {
+                                Data = data,
+                                Id = context.GetLazyString(modifier.Id),
+                                ChangeVector = modifier.ChangeVector,
+                                Flags = modifier.Flags,
+                                NonPersistentFlags = modifier.NonPersistentFlags,
+                                LastModified = modifier.LastModified ?? _database.Time.GetUtcNow(),
+                            },
+                            Attachments = null
+                        };
+                    }
+                }
+
+                foreach (var attachmentItem in storeAttachmentsForLaterMatchWithDocs)
+                {
+                    var successInIdExtraction = TryMatchAttachmentToDocumentOrStoreForLaterMatching(
+                        storeDocsForLaterMatchWithAttachments,
+                        storeAttachmentsForLaterMatchWithDocs,
+                        attachmentItem.Value,
+                        true);
+
+                    Debug.Assert(successInIdExtraction, "Already executed once so can't fail here");
+                }
+
+                foreach (var docItem in storeDocsForLaterMatchWithAttachments)
+                {
+                    if (docItem.Value.Attachments == null)
+                    {
+                        // TODO: what to do with docs reported to have attachments but attachments weren't found?
```

Comment:
What should we do here? We can reset the doc's "has attachment" flag but it feels wrong